src/main/java/seedu/address/model/assignment/DueDate.java

```diff
@@ -66,25 +77,15 @@ public DueDate(String date, String time) {
      * @param date Date of due date.
      */
     public DueDate(String date) {
-        this(date, LATEST_TIME_IN_DAY);
-    }
-
-    /**
-     * Constructs a {@code DueDate}
-     *
-     * @param friendlyDate friendly command format of date
-     * @param code code value assigned to denote friendly command
-     * @throws ParseException erroneous is given, exception is thrown
-     */
-    public DueDate(String friendlyDate, int code) throws ParseException {
-        if (code == 1) {
-            this.date = friendlyToDate(friendlyDate);
-            this.time = LocalTime.parse(LATEST_TIME_IN_DAY, PARSE_TIME_FORMAT);
-            this.dateTime = LocalDateTime.of(this.date, this.time);
-            this.value = this.dateTime.format(OUTPUT_FORMAT);
+        if (isValidFriendlyDate(date)) {
+            this.date = friendlyToDate(date);
         } else {
-            throw new ParseException(DueDate.MESSAGE_CONSTRAINTS_DUE_DATE);
+            checkArgument(isValidDate(date), MESSAGE_CONSTRAINTS_DATE);
+            this.date = LocalDate.parse(date, PARSE_DATE_FORMAT);
         }
+        this.time = LocalTime.parse(LATEST_TIME_IN_DAY, PARSE_TIME_FORMAT);
+        this.dateTime = LocalDateTime.of(this.date, this.time);
+        this.value = this.dateTime.format(OUTPUT_FORMAT);
```

Comment:
I think for this part you can just leave it as this(date, LATEST_TIME_OF_DAY) as the other constructor will do the same checks